alloy-core/src/main/java/org/starchartlabs/alloy/core/Preconditions.java

```diff
@@ -346,4 +412,68 @@ public static void checkState(boolean expression, @Nullable Supplier<String> err
         return value;
     }
 
+    /**
+     * Ensures the truth of an expression involving one or more parameters to the calling method.
+     *
+     * <p>
+     * Preferred to {@link #checkState(boolean, Object)} when elements needs to be substituted or otherwise processed
+     * for performance reasons
+     *
+     * @param expression
+     *            A boolean expression
+     * @param template
+     *            A template for a formatted message. The message is formed by replacing each {@code %s} placeholder in
+     *            the template with an argument. These are matched by position - the first {@code %s} gets
+     *            {@code args[0]}, etc. Unmatched arguments will be appended to the formatted message in square braces.
+     *            Unmatched placeholders will be left as-is.
+     * @param args
+     *            The arguments to be substituted into the message template. Arguments are converted to strings using
+     *            {@link String#valueOf(Object)}.
+     * @throws IllegalStateException
+     *             If {@code expression} is false
+     * @since 0.5.0
+     */
+    public static void checkState(boolean expression, @Nullable String template, @Nullable Object... args) {
```

Comment:
Confused, this appears to be the body of `checkArgument` with the javadoc of `checkState`, am I wrong or is something amiss here

Reply:
No, this is the correct implementation. Both `checkState` and `checkArgument` got two added implementations/overrides in this PR - for each, they have one that takes a generic boolean (expression) and a string template/subsititutions, and one that takes a predicate, value, and a string template/subsititutions. The checkState implementations should be throwing IllegalStateException, and the checkArguments IllegalArgumentException